src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -119,6 +119,27 @@ export async function testConnection(
   });
 }
 
+export async function fetchDataSourceVersion(
+  http: HttpStart,
+  { endpoint, auth: { type, credentials } }: DataSourceAttributes,
+  dataSourceID?: string
```

Comment:
should data source id be required?

Reply:
I think it is optional because of test connection. Because if we do test connection before saving data source, we won't have data source id. But what will happen if it empty or undefined? Does it throw error?

Reply:
just tested and it won't cause any exception :)  further digging indicates a maybe check`id: schema.maybe(schema.string()),` so an empty one won't cause trouble for now